libstorage/SQLBasicAccess.cpp

```diff
@@ -247,14 +247,32 @@ void SQLBasicAccess::GetCommitFieldNameAndValueEachTable(const std::string& _num
 void SQLBasicAccess::GetCommitFieldNameAndValue(const Entries::Ptr& data, const std::string& _num,
     std::map<std::string, std::vector<std::string>>& _fieldValue)
 {
-    std::map<uint64_t, std::vector<size_t>> splitDataItem;
+    std::map<std::string, std::vector<size_t>> splitDataItem;
     std::map<std::string, uint32_t> field2Score;
     uint32_t scorePower = 0;
+
+    set<std::string> keySet;
     for (size_t i = 0; i < data->size(); ++i)
     {
-        uint64_t fieldScore = 0;
         const Entry::Ptr& entry = data->get(i);
-        for (auto fieldIt : *entry)
+        for (const auto& fieldIt : *entry)

```

Comment:
The `keySet` is only used as counter, how about declare `fieldBuff` as vector and push_back directly.